tests/unit/core/sagas/test_abuse.js

```diff
@@ -0,0 +1,94 @@
+import SagaTester from 'redux-saga-tester';
+
+import * as api from 'core/api/abuse';
+import { CLEAR_ERROR } from 'core/constants';
+import abuseReducer, {
+  loadAddonAbuseReport,
+  sendAddonAbuseReport,
+} from 'core/reducers/abuse';
+import apiReducer from 'core/reducers/api';
+import abuseSaga from 'core/sagas/abuse';
+import {
+  dispatchSignInActions,
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import {
+  createFakeAddonAbuseReport,
+  createStubErrorHandler,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockApi = sinon.mock(api);
+    const initialState = dispatchSignInActions().state;
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: {
+        abuse: abuseReducer,
+        api: apiReducer,
+      },
+    });
+    sagaTester.start(abuseSaga);
+  });
+
+  function _sendAddonAbuseReport(params) {
+    sagaTester.dispatch(sendAddonAbuseReport({
+      addonSlug: fakeAddon.slug,
+      errorHandler: createStubErrorHandler('my-abuse-error'),
+      message: 'Testing',
+      ...params,
+    }));
+  }
+
+  it('calls the API for abuse', async () => {
+    const addon = { ...fakeAddon, slug: 'fancy' };
```

Comment:
I don't think this should inherit from `...fakeAddon` since that will include a lot more fields than what's returned from the API.

Reply:
I just checked and this uses `createFakeAddonAbuseReport` which only returns the relevant parts of the add-on in the response. So this is okay.